src/components/file-browser/file-browser-hooks.test.ts

```diff
@@ -0,0 +1,110 @@
+import { act, renderHook } from '@testing-library/react-hooks'
+import _ from 'lodash/fp'
+import { IncrementalResponse } from 'src/components/file-browser/file-browser-backends'
+import { useIncrementalResponse } from 'src/components/file-browser/file-browser-hooks'
+
+
+describe('useIncrementalResponse', () => {
+  // Returns an incremental response with 3 pages of 3 numbers each
+  const getTestIncrementalResponse = (): Promise<IncrementalResponse<number>> => {
+    const getNextPage = (previousResults: number[], pageNumber: number): Promise<IncrementalResponse<number>> => {
+      const results = [...previousResults, ..._.range(pageNumber * 3 + 1, (pageNumber + 1) * 3 + 1)]
+      const hasNextPage = pageNumber < 2
+      return Promise.resolve({
+        results,
+        getNextPage: hasNextPage ?
+          () => getNextPage(results, pageNumber + 1) :
+          () => { throw new Error('No next page') },
+        hasNextPage
+      })
+    }
+
+    const firstPageResults = [1, 2, 3]
+    return Promise.resolve({
+      results: firstPageResults,
+      getNextPage: () => getNextPage(firstPageResults, 1),
+      hasNextPage: true
+    })
+  }
+
+  it('gets initial response', async () => {
+    const { result, waitForNextUpdate } = renderHook(() => useIncrementalResponse(getTestIncrementalResponse))
+    expect(result.current.results).toEqual([])
+    await waitForNextUpdate()
+    expect(result.current.results).toEqual([1, 2, 3])
```

Comment:
it would be really nice if these result properties were 'unwrapped' in the hook's return to not require the .current in the object/prop pathing on all of these usages.

Reply:
They can't be fully destructured there. `result.current` is like a ref and mutable. So the destructured value would be out of date after an update.

So in this case:
```ts
const { result, result: { current: { results } }, waitForNextUpdate } = renderHook(() => useIncrementalResponse(getTestIncrementalResponse))
expect(result.current.results).toEqual([]) // results = []
await waitForNextUpdate()
expect(result.current.results).toEqual([1, 2, 3]) // results = []

Reply:
More information about this in React Hooks Testing Library's documentation.
https://react-hooks-testing-library.com/reference/api#renderhook-result

Reply:
Discussed in a meeting... our conclusion was that renaming `result` makes this clearer.